lib/components/button-bar-ai-popover/index.tsx

```diff
@@ -0,0 +1,187 @@
+import { Popover } from "@mui/material";
+import React, {
+  ComponentProps,
+  FC,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import ButtonBarAI from "../button-bar-ai";
+import { ActionBarAIButtonName } from "@appflowy-chat/types";
+import IconCheck from "@appflowy-chat/assets/icons/check.svg?react";
+import IconRegenerate from "@appflowy-chat/assets/icons/regenerate.svg?react";
+import "./index.css";
+import { ActionBarAIButtonDataPopover } from "@appflowy-chat/utils/actionBarAIButtons";
+import TooltipDefault from "../tooltip-default";
+
+interface Props extends ComponentProps<"button"> {
+  icon: ReactNode;
+  tooltip: string;
+  optionsData: ActionBarAIButtonDataPopover["optionsData"];
+  activeOption: string;
+  btnName: ActionBarAIButtonName;
+  onOptionChange: (value: string) => void;
+}
+
+const ButtonBarAIPopover: FC<Props> = ({
+  icon,
+  tooltip,
+  optionsData,
+  activeOption,
+  onOptionChange,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+  const [shortOptionName, setShortOptionName] = useState<string>("");
+
+  const rootRef = useRef<HTMLDivElement>(null);
+
+  function handleClick(e: MouseEvent<HTMLButtonElement>) {
+    setIsOpen(true);
+    setAnchorEl(e.currentTarget);
+    handlePopoverStateChange("open");
+  }
+  function handleClose() {
+    setIsOpen(false);
+    setAnchorEl(null);
+    setTimeout(() => {
+      handlePopoverStateChange("close");
+    }, 300);
+  }
+
+  function handlePopoverStateChange(action: "open" | "close") {
+    const messageCard =
+      rootRef?.current?.closest<HTMLDivElement>(".message-ai");
```

Comment:
Maybe you can use context to manipulate the state, making it easier to control data and trace its source.

Reply:
I added the state manipulation instead of storing isPopoverOpen state in DOM.